cooltools/snipping.py

```diff
@@ -361,11 +490,28 @@ def select(self, region1, region2):
         )
         if self.cooler_opts["sparse"]:
             matrix = matrix.tocsr()
-        self._isnan1 = np.isnan(self.clr.bins()["weight"].fetch(region1_coords).values)
-        self._isnan2 = np.isnan(self.clr.bins()["weight"].fetch(region2_coords).values)
+        if self.clr_weight_name:
+            self._isnan1 = np.isnan(
+                self.clr.bins()[self.clr_weight_name].fetch(region1_coords).values
+            )
+            self._isnan2 = np.isnan(
+                self.clr.bins()[self.clr_weight_name].fetch(region2_coords).values
+            )
+        else:
+            self._isnan1 = np.zeros_like(
```

Comment:
mybe something like:
```python
l,h = clr.extent(region1_coords)
self._isnan1 = np.zeros(h-l)
l,h = clr.extent(region2_coords)
self._isnan2 = np.zeros(h-l)
```

maybe some tiny functions that tells how many bins are there in a region is needed (bioframe, cooler, cooltools ?)

Reply:
not important at all, just a thought